tests/ui/perfherder/compare-view/compare_table_test.jsx

```diff
@@ -684,3 +685,35 @@ test(`table data sorted by 'Confidence' has data with invalid confidence at the
   expect(compareTableRows[1]).toContainElement(result2);
   expect(compareTableRows[2]).toContainElement(result3);
 });
+
+const propsWithUnit = {
+  onChangeSort: jest.fn(),
+  column: {
+    name: 'Base',
+    currentSort: 'default',
+  },
+  measurementUnit: 'score',
+};
+
+const propsWithoutUnit = {
+  onChangeSort: jest.fn(),
+  column: {
+    name: 'New',
+    currentSort: 'default',
+  },
+};
+
+test(`measurement unit should appear only when passed as props`, async () => {
+  const { queryByText, rerender } = render(
+    <CompareSortButton {...propsWithUnit} />,
+  );
+
+  expect(queryByText('Base (score)')).not.toBeNull();
+  expect(queryByText('Base (score)')).toBeTruthy();
+
+  rerender(<CompareSortButton {...propsWithoutUnit} />);
+
+  expect(queryByText('New (score)')).not.toBeInTheDocument();
+  expect(queryByText('New (score)')).toBeFalsy();
```

Comment:
A single expect should suffice. As long as the string is found in the DOM, we should be OK.